python/pyspark/sql/tests/test_udtf.py

```diff
@@ -3490,39 +3490,38 @@ def eval(self):
                     udtf(TestUDTF, returnType=ret_type)().collect()
 
 
-def test_udtf_with_collated_string_types(self):
-    @udtf(
-        "out1 string, out2 string collate UTF8_BINARY, out3 string collate UTF8_LCASE,"
-        " out4 string collate UNICODE"
-    )
-    class MyUDTF:
-        def eval(self, v1, v2, v3, v4):
-            yield (v1 + "1", v2 + "2", v3 + "3", v4 + "4")
-
-    schema = StructType(
-        [
-            StructField("col1", StringType(), True),
-            StructField("col2", StringType("UTF8_BINARY"), True),
-            StructField("col3", StringType("UTF8_LCASE"), True),
-            StructField("col4", StringType("UNICODE"), True),
+    def test_udtf_with_collated_string_types(self):
```

Comment:
I think this test should be included in `BaseUDTFTestsMixin`, because `UDTFArrowTestsMixin` is specific for the configs:
```
        cls.spark.conf.set("spark.sql.execution.pythonUDTF.arrow.enabled", "true")
        cls.spark.conf.set(
            "spark.sql.legacy.execution.pythonUDTF.pandas.conversion.enabled", "false"
        )
```